src/main/java/hillel/spring/doctor/DoctorRestAPIController.java

```
@@ -0,0 +1,99 @@
+                                   SpecializationsConfig specializationsConfig,
+                                   @Value("${doctor.hostname:localhost}") String hostname) {
+        this.doctorRestAPIService = doctorRestAPIService;
+        this.doctorDtoConverter = doctorDtoConverter;
+        this.specializationsConfig = specializationsConfig;
+        this.uriComponentsBuilder = UriComponentsBuilder.newInstance()
+                .scheme("http")
+                .host(hostname)
+                .port(8081)
+                .path("/doctors/{id}");
+    }
+
+    @GetMapping("/doctors")
+    public List<DoctorOutputDto> findDoctors(@RequestParam Optional<String> specialization,
+                                             @RequestParam Optional<String> name,
+                                             @RequestParam Optional<List<String>> specializations){
+
+        return doctorRestAPIService.findDoctors(specialization, name, specializations).stream()
+                                                          .map(doctorDtoConverter::toDto)
+                                                          .collect(Collectors.toList());
+    }
+
+    @GetMapping("/doctors/{id}")
+    public DoctorOutputDto findDoctorByID(@PathVariable Integer id){
+        val mayBeDoctor = doctorRestAPIService.findDoctorByID(id);
+
+        return doctorDtoConverter.toDto(mayBeDoctor.orElseThrow(DoctorNotFoundException::new));
+    }
+
+    @PostMapping("/doctors")
+    public ResponseEntity<?> createDoctor(@RequestBody DoctorInputDto docDto){
+        final val doctor = doctorDtoConverter.toModel(docDto);
+
+        if(checkSpecialization(doctor)) {
+            try {
+                final val newDoc = doctorRestAPIService.createDoctor(doctor);
+                return ResponseEntity.created(uriComponentsBuilder.build(newDoc.getId())).build();
+            } catch (Exception e) {
+                throw new RuntimeException("Oops, error");
```

It is a really bad idea to swallow original exception: it contains a message and a stack trace. They can help find out what gone wrong.
Another bad idea to use this "Oops, error" it says nothing about the root cause, and it is better to avoid such messages.
So you should either wrap original exception or use lombok's `@SneakyThrows`  annotation